src/orquestra/sdk/_base/serde.py

```diff
@@ -34,11 +35,28 @@ def _chunkify(s: str) -> t.List[str]:
 def _encoded_pickle_chunks(object: t.Any) -> t.List[str]:
     return _chunkify(
         codecs.encode(
-            dill.dumps(object, protocol=PICKLE_PROTOCOL, recurse=True), ENCODING
+            cloudpickle.dumps(object, protocol=PICKLE_PROTOCOL), ENCODING
         ).decode()
     )
 
 
+@contextmanager
+def registered_module(module):
+    if module is not None:
+        register_pickle_by_value(module)
+    yield
+    if module is not None:
+        unregister_pickle_by_value(module)
```

Comment:
**issue**: `unregister_` won't be called if `yield` throws an exception. [More info](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager).

```suggestion
    try:
        yield
    finally:
        if module is not None:
            unregister_pickle_by_value(module)
```

Reply:
Good spot. You can also check out this [helpful blog post](https://jamesclark.dev/context-managers/#helper-decorator) for more info. :^)